### PR TITLE
Todo list for peer to peer code review

### PR DESCRIPTION
**Pull request by julie-ify**
Here is the copy of the Todo List App I created for peer to peer code review.

This pull request is to check the quality of the code using the following:
JavaScript best practices.
DRY, KISS, and YAGNI rules.
HTML & CSS best practices.

### CODE REVIEW

**Comment by medaminedev66** [comment]
Hi @julie-ify,

Good job so far! your work is awesome

## Key Highlights
- your code is clean,
- your code organization was really good. 

## Required Changes 
Check the comments under the review.

## Optional suggestion
In `index.html` file:
The type of button in the form is 'submit'.  For a better user experience, and to make the web application more dynamically, it would be better to use the 'button' type to not make the page reloaded after clicking on the button.

File: src/addlist.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/* eslint-disable import/no-cycle */
+
+import { saveStorage } from './storage.js';
+import populateList from './index.js';
+
+const addNewTask = (input) => {
+  const taskList = localStorage.getItem('todo-list')
+    ? JSON.parse(localStorage.getItem('todo-list'))
+    : [];
```
[the remark this answers is not in the document]
Comment (julie-ify) [reply src/addlist.js:9]: Exactly, I will do that now.

File: src/edit.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable import/no-cycle */
+
+import populateList from './index.js';
+import { saveStorage } from './storage.js';
+
+const editTask = (text, tasks, task) => {
+  const taskIndex = task.index;
```
[the remark this answers is not in the document]
Comment (julie-ify) [reply src/edit.js]: I have worked on this too.

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+const populateList = () => {
+  while (listContainer.lastChild) {
+    listContainer.removeChild(listContainer.lastChild);
+  }
+
+  const tasks = getStorage();
+
+  if (tasks != null) {
+    for (let i = 0; i < tasks.length; i += 1) {
```
[the remark this answers is not in the document]
Comment (julie-ify) [reply src/index.js]: I have replaced the for loop with forEach as requested.

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+      trash.innerHTML = "<i class='fas fa-trash-alt'></i>";
+      trash.style.display = 'none';
+      trash.id = tasks.indexOf(tasks[i]);
+
+      span.appendChild(dot);
+      list.appendChild(listFChild);
+      listFChild.appendChild(input);
+      listFChild.appendChild(label);
+      listFChild.appendChild(span);
+      listFChild.appendChild(trash);
```
[the remark this answers is not in the document]
Comment (julie-ify) [reply src/index.js]: This has been taking care of by using forEach.

File: src/trashTask.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+  const currentTasks = storedTasks.filter((index) => storedTasks.indexOf(index) !== taskIndex);
+
+  for (let i = 0; i < currentTasks.length; i += 1) {
+    currentTasks[i].index = currentTasks.indexOf(currentTasks[i]) + 1;
+  }
+
+  saveStorage(currentTasks);
+
+  window.location.reload();
```
[the remark this answers is not in the document]
Comment (julie-ify) [reply src/trashTask.js:14]: Ok, Thanks

**Comment by medaminedev66** [comment]
## Hi @julie-ify, 

Good job making the changes required.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

Cheers and Happy coding!👏👏

**Pull request merged**